src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.chat.util;
+
+import java.io.IOException;
+
+public final class Tokenizer {
+  private StringBuilder token;
+  private String source;
+  private int at;
+
+  public Tokenizer(String source) {
+	this.source = source;
+	at = 0;
+	token = new StringBuilder();
+  }
+
+  public String next() {
+    try {
+      // Skip all leading whitespace
+      while (remaining() > 0 && Character.isWhitespace(peek())) {
+        read();  // ignore the result because we already know that it is a whitespace character
+      }
+      if (remaining() <= 0) {
+        return null;
+      } else if (peek() == '"') {
+        return readWithQuotes();
+      } else {
+        return readWithNoQuotes();
+      }
+    } catch(IOException e) {
+      return "An IOException was thrown in Tokenizer.next()";
+    }
+  }
+
+  private int remaining() {
+		return source.length() - at;
```

Comment:
Don't mix tabs and spaces. This codebase generally uses spaces, so I would stick to that.